lib/core/form/components/widgets/text/text.widget.spec.ts

```diff
@@ -75,14 +81,124 @@ describe('TextWidgetComponent', () => {
+                widget.field = new FormFieldModel(new FormModel({ taskId: 'fake-task-id' }), {
+                    id: 'text-id',
+                    name: 'text-name',
+                    value: '',
+                    type: FormFieldTypes.TEXT,
+                    readOnly: false
+                });
+
+                fixture.detectChanges();
+                const textWidgetLabel = element.querySelector('label');
+                expect(textWidgetLabel.innerText).toBe('text-name');
+            });
+
+            it('should be able to set a Text Widget as required', async () => {
+                widget.field = new FormFieldModel(new FormModel({ taskId: 'fake-task-id' }), {
+                    id: 'text-id',
+                    name: 'text-name',
+                    value: '',
+                    type: FormFieldTypes.TEXT,
+                    readOnly: false,
+                    required: true
+                });
+
+                fixture.detectChanges();
+                const textWidgetLabel = element.querySelector('label');
+                expect(textWidgetLabel.innerText).toBe('text-name*');
+                expect(widget.field.isValid).toBeFalsy();
+
+                inputElement = <HTMLInputElement> element.querySelector('#text-id');
+                inputElement.value = 'TEXT';
+                inputElement.dispatchEvent(new Event('input'));
+                await fixture.whenStable();
+                expect(widget.field.isValid).toBeTruthy();
+
+                inputElement.value = '';
+                inputElement.dispatchEvent(new Event('input'));
+                await fixture.whenStable();
+                expect(widget.field.isValid).toBeFalsy();
+            });
+
+            it('should be able to set a placeholder for Text widget', () => {
+                widget.field = new FormFieldModel(new FormModel({ taskId: 'fake-task-id' }), {
+                    id: 'text-id',
+                    name: 'text-name',
+                    value: '',
+                    type: FormFieldTypes.TEXT,
+                    readOnly: false,
+                    placeholder: 'Your name here'
+                });
+                fixture.detectChanges();
+                const textWidgetLabel = element.querySelector('input');
+                expect(textWidgetLabel.getAttribute('placeholder')).toBe('Your name here');
+            });
+
+            it('should be able to set min/max length properties for Text widget', async () => {
+                widget.field = new FormFieldModel(new FormModel({ taskId: 'fake-task-id' }), {
+                    id: 'text-id',
+                    name: 'text-name',
+                    value: '',
+                    type: FormFieldTypes.TEXT,
+                    readOnly: false,
+                    minLength: 5,
+                    maxLength: 10
+                });
+                fixture.detectChanges();
+                inputElement = <HTMLInputElement> element.querySelector('#text-id');
```

Comment:
if the input typing operation is repeated it might be a good idea to put them in a single function.